### ISSUE TITLE
Make primitive types in prims as eqtype

### ISSUE THREAD

**Issue by aseemr** [FStarLang/FStar#1514]
We generate unnecessary SMT queries for `hasEq int`, `hasEq bool`, `hasEq nat`, etc. I tried marking these as `eqtype` in `prims`, but got some regressions. Will try to squash it in this cycle.

**Comment by mtzguido** [comment]
Another option we discussed in the call is simplifying `hasEq` in the normalizer. For instance, adding reduction such as:

```
hasEq int -> True
hasEq (x:int{_}) -> True
hasEq nat -> True
hasEq (list a) -> hasEq a
```

I find it a bit less elegant than the annotating in prims, but has the advantage that it could reduce `hasEq (list int)` to `True`, which annotations won't do. On the other hand, I think this would fail:
```
let test (a:Type) (_ : squash (hasEq (list a))) =
    assert (hasEq (list a))
```
since the assertion would be simplified to `hasEq a`, and there's no way of proving. Probably not too serious in practice, but worth keeping in mind.

**Comment by mtzguido** [comment]
I have Aseem's version of the fix in the PR above, and Aseem has my version on a branch of his. I messed up and forgot to sync with him before starting to work on this, but seems we got lucky :).

**Comment by aseemr** [comment]
I pushed the normalizer changes to master. Since it changes the wps, also made sure that there is no verification impact on Everest.

**Comment by mtzguido** [comment]
They are now also `eqtype`s in prims, closing.
